Adding PyPy support (#185)
joerick/cibuildwheel: docs/options.md

`auto` will auto-detect platform using environment variables, such as `TRAVIS_OS_NAME`/`APPVEYOR`/`CIRCLECI`.

> For `linux` you need Docker running, on macOS or Linux. For `macos`, you need a Mac machine, and note that this script is going to automatically install MacPython on your system, so don't run on your development machine. For `windows`, you need to run in Windows, and it will build and test for all versions of Python under `C:\cibw\python`.

Reviewer: This seems to suggest the versions need to be preinstalled at this location.

```suggestion
For `linux` you need Docker running, on macOS or Linux. For `macos`, you need a Mac machine, and note that this script is going to automatically install MacPython on your system, so don't run on your development machine. For `windows`, you need to run in Windows, and cibuildwheel will install required versions of Python to `C:\cibw\python` using nuget.
```

| Python 2.7      | cp27-macosx_x86_64  | cp27-manylinux_x86_64  | cp27-manylinux_i686  | cp27-win_amd64  | cp27-win32     |
| Python 3.5      | cp35-macosx_x86_64  | cp35-manylinux_x86_64  | cp35-manylinux_i686  | cp35-win_amd64  | cp35-win32     |
| Python 3.6      | cp36-macosx_x86_64  | cp36-manylinux_x86_64  | cp36-manylinux_i686  | cp36-win_amd64  | cp36-win32     |
| Python 3.7      | cp37-macosx_x86_64  | cp37-manylinux_x86_64  | cp37-manylinux_i686  | cp37-win_amd64  | cp37-win32     |
| Python 3.8      | cp38-macosx_x86_64  | cp38-manylinux_x86_64  | cp38-manylinux_i686  | cp38-win_amd64  | cp38-win32     |
| PyPy 2.7 v7.3.0 | pp27-macosx_x86_64  | pp27-manylinux_x86_64  |                      |                 | pp27-win32     |
| PyPy 3.6 v7.3.0 | pp36-macosx_x86_64  | pp36-manylinux_x86_64  |                      |                 | pp36-win32     |

The list of supported and currently selected build identifiers can also be retrieved by passing the `--print-build-identifiers` flag to `cibuildwheel`.
The format is `python_tag-platform_tag`, with tags similar to those in [PEP 425](https://www.python.org/dev/peps/pep-0425/#details).

> For CPython, the minimally supported OS X/macOS version is 10.9; for PyPy 2.7 and PyPy 3.6, respectively macOS 10.7 and 10.13 or higher is required.

Reviewer: ```suggestion
For CPython, the minimally supported macOS version is 10.9; for PyPy 2.7 and PyPy 3.6, respectively macOS 10.7 and 10.13 or higher is required.
```

# Only build on Python 3 and skip 32-bit builds
CIBW_BUILD: cp3?-*
CIBW_SKIP: "*-win32 *-manylinux_i686"

# Only build PyPy and CPython 3

> CIBW_BUILD: pp*-* cp3?-*

Reviewer 2: Why `pp*-*` not `pp*`?

Author: Because ... well, because the PyPy tags were changed a few times, and because above it also says `cp3?-*` rather than `cp3*`; no real further reason. Do you think `pp*` is more clear?

Reviewer 2: Such writing suggest that there are some tags starting with `pp` without `-`.

Author: Can you not say the same about `cp3?-*`, then, that it suggests there are some tags with `cp3*` that are not `cp3?-*`?

Reviewer 2: For me there is a big difference between one character match`?` and wildcard `*`. Maybe this example should show pypy3 and cPython3.

Btw. see line 114 This Python should be renamed to cPython?

# Only build on Python 3 and skip 32-bit builds
CIBW_BUILD: cp3?-*
CIBW_SKIP: "*-win32 *-manylinux_i686"

# Only build PyPy and CPython 3

> CIBW_BUILD: pp* cp3?-*

Reviewer: Maybe also worth adding a skip PyPy example e.g.

```yaml
# Disable building PyPy wheels on all platforms
CIBW_SKIP: pp*
```

Author: OK, will add this tomorrow.

Author: Not sure where to add it, but it's now just a the bottom of the list.

Note that `auditwheel` detects the version of the `manylinux` standard in the Docker image through the `AUDITWHEEL_PLAT` environment variable, as `cibuildwheel` has no way of detecting the correct `--plat` command line argument to pass to `auditwheel` for a custom image. If a Docker image does not correctly set this `AUDITWHEEL_PLAT` environment variable, the `CIBW_ENVIRONMENT` option can be used to do so (e.g., `CIBW_ENVIRONMENT='AUDITWHEEL_PLAT="manylinux2010_$(uname -m)"'`).

Note that `manylinux2014` doesn't support builds with Python 2.7 - when building with `manylinux2014`, skip Python 2.7 using `CIBW_SKIP` (see example below).

#### Examples

```yaml
# build using the manylinux1 image to ensure manylinux1 wheels are produced
# skip PyPy, since there is no PyPy manylinux1 image
CIBW_MANYLINUX_X86_64_IMAGE: manylinux1
CIBW_MANYLINUX_I686_IMAGE: manylinux1

> CIBW_SKIP: pp??_??-*

Reviewer: ```suggestion
CIBW_SKIP: pp*
```
...maybe simpler, no?

Author: The main idea was to mimick the `CIBW_BUILD: cp3?-*` example. I'm happy changing it again as well; as you prefer.

Author: Changed to `pp*-*` to keep a bit of consistency.
